lib/uhash.ts

```diff
@@ -0,0 +1,54 @@
+// This module implements the "uhash" signature specification described in docs/signature_specification.md
+import * as crypto from 'crypto';
+import { SignatureResult } from './types';
+import { isBinary } from './utils/binary';
+
+const hashAlgorithm = 'md5';
+
+export const getUhashSignature = (
+  filePath: string,
+  fileContents: Buffer,
+): SignatureResult => {
+  const digest = getUhashDigest(fileContents, isBinary(fileContents));
+  const data = digest.slice(0, 24);
+  return {
+    path: filePath,
+    hashes_ffm: [
+      {
+        data,
+        format: 3,
+      },
+    ],
+  };
+};
+
+const getUhashDigest = (fileContents: Buffer, isBinary: boolean): string => {
+  const file = isBinary ? fileContents : removeWhitespaceBytewise(fileContents);
+  if (hashAlgorithm == 'md5') {
```

Comment:
Maybe a matter of taste, but I would prefer to early exit by reverting the if else order, and then remove the else.

```typescript
const file = isBinary ? fileContents : removeWhitespaceBytewise(fileContents);

if (hashAlgorithm !=== 'md5') {
  throw new Error(`hashAlgorithm ${hashAlgorithm} is not supported`);
}

const hash = crypto.createHash(hashAlgorithm).update(file);
return hash.digest('hex');
```

Since file is only used in your if-block, maybe we could rewrite it to only run if it's the correct hashAlgorithm

```typescript
if (hashAlgorithm !=== 'md5') {
  throw new Error(`hashAlgorithm ${hashAlgorithm} is not supported`);
}

const file = isBinary ? fileContents : removeWhitespaceBytewise(fileContents);

const hash = crypto.createHash(hashAlgorithm).update(file);
return hash.digest('hex');
```

Reply:
fixed in latest version.